### ISSUE TITLE
Add `Granite::Migrator#create_sql` and `#drop_sql`

### ISSUE THREAD

**Issue by Jens0512** [amberframework/granite#319]
### tl;dr

`Granite::Migrator` generates SQL on `#drop` and `#create`, and runs it imminently. The possibility of having the migrator export the generated SQL would be nice for use in migration tools like `micrate`. The generating functionality is already in place, implementing this would just be moving the generating code into `create`- and `drop`- `_sql` methods (names debatable), and using them in place of the current generating code.

---

I'm using Granite for building a program (i.e. no amber), I'm also quite new to SQL, which is one of the reasons I enjoy Granite so much. 

Following the Granite [docs on migrating][docs], I stumble upon SQL in the micrate up and down sections, I do not know how Granite interfaces with its tables. Therefore I do not know how to define these migrations, and I take a look at the alternative `Granite::Migrator#create` and `#drop`, which both work great, however they seem wholly impractical to use long term, at least compared to `micrate`. 

I hoped I could extract the SQL generated by `Granite::Migrator`, but seems that's not possible, hence I ask here whether it's a good idea or not to extract the generated code into separate methods. Practical or not it adheres better to the single responsibility principle... Sorry for the spaghetti text.

I'll happily implement this if it's accepted as a good idea.

[docs]: https://github.com/amberframework/granite/blob/master/docs/migrations.md#creating-a-migration

**Comment by Jens0512** [comment]
`#generate_drop_sql` may be a better name.

**Comment by robacarp** [comment]
@Jens0512 I think this is a fine idea.

**Comment by Jens0512** [comment]
Great, I can't get the specs working (docker image problems), but I got this done locally. Shall I still just PR?

**Comment by robacarp** [comment]
@Jens0512 a PR is welcome, yep.
